pynxtools/dataconverter/template.py

```diff
@@ -125,6 +125,13 @@ def __contains__(self, k):
             k in self.required
         ])
 
+    def get(self, key: str, default=None):
+        """Proxies the get function to our internal __getitem__"""
+        try:
+            return self[key]
+        except KeyError:
+            return default
+
     def __getitem__(self, k):
```

Comment:
@sherjeelshabih I tried it out on the day before yesterday.
Here I added the code output:
```ipython
template = Template()
In [18]: generate_template_from_nxdl(nxdl_root, template)

In [19]: template['hl']
Out[19]: 
{'/ENTRY[entry]/title': None,
 '/ENTRY[entry]/definition': None,
 '/ENTRY[entry]/start_time': None,
 '/ENTRY[entry]/definition/@version': None,
 '/ENTRY[entry]/USER[user]/name': None,
 '/ENTRY[entry]/USER[user]/email': None,
 '/ENTRY[entry]/INSTRUMENT[instrument]/energy_resolution': None,
 '/ENTRY[entry]/INSTRUMENT[instrument]/energy_resolution/@units': None,
 '/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/type': None,
 '/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/probe': None,
 '/ENTRY[entry]/INSTRUMENT[instrument]/SOURCE[source]/name': None,
 '/ENTRY[entry]/INSTRUMENT[instrument]/BEAM[beam]/distance': None,
 ```
template['hl'] expected to raise an ValueError.

Reply:
Yes, it is not expected to return a value for a key that is not even exist. so it should return an error, KeyError not an irrelevant dict.